### PR TITLE
Annotation | Label Search

### PR DESCRIPTION
**Pull request by stalberm**
## Description
* Replaces the old popup that prompted manually typing a label with a label search
* Adds json to contain all possible labels, search options are populated using this json
* Adds option to add label if it's not found in the provided list

### CODE REVIEW

File: lib/pages/detection/annotation.dart [hunk 1]
```diff
@@ -1,18 +1,20 @@
...
 import 'package:flutter/rendering.dart';
+import 'package:flutter/services.dart';
 
 // Package imports:
 import 'package:go_router/go_router.dart';
+import 'package:multiple_search_selection/createable/create_options.dart';
 import 'package:provider/provider.dart';
 import 'package:shared_preferences/shared_preferences.dart';
-
+import 'package:multiple_search_selection/multiple_search_selection.dart';
```
Comment (bpugh5) [inline lib/pages/detection/annotation.dart:14]: little itty bitty thing, could you slide into the imports in alphabetical order?

File: lib/pages/detection/annotation.dart [hunk 2]
```diff
@@ -262,15 +280,25 @@ class _AnnotationPageState extends State<AnnotationPage> {
...
-                              notifier.label == null
-                                  ? "Add Label"
-                                  : "Change Label",
+                              "Select Label",
+                              style: textTheme.labelLarge!.copyWith(
+                                color: Theme.of(context).colorScheme.onPrimary,
+                              ),
                             ),
+                            onPressed: () {
+                              labels.isNotEmpty
```
Comment (bpugh5) [inline lib/pages/detection/annotation.dart:290]: could you comment this logic briefly just to describe what it does onPressed?

**Review by bpugh5 — COMMENTED** [review]
See comments! Nothing bad, but little things I noticed, otherwise looks good :)

File: .vscode/launch.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{
```
Comment (kimberlymarkley) [inline .vscode/launch.json:1]: I poked around online to see what this file is for, but couldn't figure out why we need it. Is it for versioning? or???

File: lib/pages/detection/annotation.dart [hunk 4]
```diff
@@ -385,3 +413,98 @@ class _AnnotationPageState extends State<AnnotationPage> {
     );
   }
 }
+
```
Comment (kimberlymarkley) [inline lib/pages/detection/annotation.dart:416]: This code is super easy to read. Who says CS folks can't name stuff thoughtfully?!  :D

**Review by kimberlymarkley — APPROVED** [review]

File: .vscode/launch.json (hunk 3 above)
Comment (stalberm) [reply .vscode/launch.json:1]: I really am not sure. I don't know when that got added sadly.

Comment (theusaf) [reply .vscode/launch.json:1]: Might be worth removing and/or adding to gitignore. This file is generally for developer-specific configurations

File: lib/pages/detection/annotation.dart [hunk 5]
```diff
@@ -45,9 +50,20 @@ class _AnnotationPageState extends State<AnnotationPage> {
   @override
   void initState() {
     super.initState();
+    readJson();
     initPreferences();
   }
 
+  /// Reads json file containing the possible labels
+  Future<void> readJson() async {
```
Comment (theusaf) [inline lib/pages/detection/annotation.dart:58]: I think a better name could be `loadLabels` or something

File: lib/util/providers.dart [hunk 6]
```diff
@@ -1,4 +1,5 @@
 // Flutter imports:
+import 'package:binsight_ai/widgets/free_draw.dart';
```
Comment (theusaf) [inline lib/util/providers.dart:2]: Should probably be placed in its original location

File: lib/widgets/free_draw.dart [hunk 7]
```diff
@@ -1,8 +1,6 @@
 // Flutter imports:
 import 'dart:ui' as ui;
 import 'package:flutter/material.dart';
-
-// Package imports:
```
Comment (theusaf) [inline lib/widgets/free_draw.dart:5]: To be consistent with other files, we should leave this in

**Review by theusaf — CHANGES_REQUESTED** [review]

**Review by kimberlymarkley — DISMISSED** [review]

**Review by bpugh5 — APPROVED** [review]
Looks great!!

File: lib/widgets/free_draw.dart [hunk 8]
```diff
@@ -1,9 +1,8 @@
...
 import 'package:provider/provider.dart';
+// Package imports:
```
Comment (theusaf) [inline lib/widgets/free_draw.dart:5]:
```suggestion

// Package imports:
import 'package:provider/provider.dart';
```

**Review by theusaf — APPROVED** [review]

**Pull request merged**